### ISSUE TITLE
The latest version of Azure Storage Explorer (1.29.1) is unable to view data in the Azure Storage Account table.

### ISSUE THREAD

**Issue by takuya-shima** [microsoft/AzureStorageExplorer#6896]
### Preflight Checklist

- [X] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.29.1

### Regression From

1.28.0

### Architecture

i86

### Storage Explorer Build Number

April 2023 (Version 1.29.1, build 20230509.5)

### Platform

Windows

### OS Version

Windows 10

### Bug Description

The latest version of Azure Storage Explorer (1.29.1) is unable to view data in the Azure Storage Account table; narrowing down the data to be viewed by timestamp does not allow the most recent data to be viewed. We have confirmed with several members that this event is reproducible. In previous versions(1.28.0), of course, we were able to view the data of Azure Storage Account. This latest version (1.29.1) have a bug and occures degression. It is not available for very commercial use.

### Steps to Reproduce

1.Launch Azure Storage Explorer.
2.Select any Azure Storage Account.
3.Select any table.
4.Run a query to see the data in the table. Now filter the data by timestamp. This can be custom or the last 24 hours.
5.Once you have selected the data, execute the query.
6.No data will appear after the query is executed.

### Actual Experience

This event occurs when narrowing down data by timestamp when executing a query. You may be able to view the data if you narrow down the data by other fields.
However, if you want to view the most recent data, you have no choice but to narrow down the data by timestamp, so this event is clearly a bug.

### Expected Experience

In 1.29.1, when browsing an Azure Storage Account, it is not possible to run a query narrowing the data by timestamp; in 1.28.0, it was possible. This is clearly a bug and is causing a migration.

### Additional Context

The latest version (1.29.1) is not available in commercial environment. 
We cannot view the data of Azure Storage Account due to bug. 
Please revise it immediately. 
Or, please announce to use 1.28.0.

**Comment by craxal** [comment]
@takuya-shima I'm afraid I'm not able to reproduce the issue you're describing. I have several follow-up questions to get more info:

1. Can you please share the query you are running here? Just save your query and attach the query file here. Knowing the exact query you're running may help us better understand the problem you're facing.
2. Are you using the query builder (UI for building queries) or the query editor (text field that lets you enter OData query syntax directly)?
3. If you are using the query builder, which timestamp options are you using? Are you using one of the relative options (Last 7 Days) or a custom range?
4. If you are using the query builder and you switch to the query editor, what is the query string that you find?
5. Are you getting any results at all if you run your query?
6. If you switch to the query editor and run the query, do you get any error messages?

**Comment by takuya-shima** [comment]
We are cloud application development engineer for Olympus Japan. 
Our team develops various applications on Microsoft Azure. 
Our team has confirmed that this defect occurs and is reproducible. 
This bug occurs on the query builder of Version 1.29.1


- This bug detail of Version 1.29.1
For example, after connecting to the storage account, set timestamp to the last 24 hours on the query builder and run the query.

<img width="960" alt="Azure Storage Explorer 1 29 1" src="https://github.com/microsoft/AzureStorageExplorer/assets/104060952/5a6beaa6-e867-4dc7-94a2-813b942b48b8">



// Your Comment and my Answer

1. Can you please share the query you are running with us? Just save your query and attach the query file in a comment. Knowing the exact query you're running may help us better understand the problem you're facing.

- For example, after connecting to the storage account, set timestamp to the last 24 hours on the query builder and run the query.
The result of the query will return "No data to display. 
Of course, there is plenty of data for the last 24 hours in the storage account, and it does indeed show up in 1.28.0.


<img width="960" alt="Azure Storage Explorer 1 28 0" src="https://github.com/microsoft/AzureStorageExplorer/assets/104060952/0706267d-adcf-44ce-bfde-21a6dc48f4fb">


2. Are you using the query builder (UI for building queries) or the query editor (text field that lets you enter OData query syntax directly)?

- We use a query builder. This bug occurs on the query builder.


3. If you are using the query builder, which timestamp options are you using? Are you using one of the relative options (Last 7 Days) or a custom range?

- This bug occurs in both cases, whether relative or custom options. It occurs even if the last 7 days are specified, or if any time period is specified in the custom option.

<img width="960" alt="Azure Storage Explorer 1 29 1 -2" src="https://github.com/microsoft/AzureStorageExplorer/assets/104060952/f8cc0f51-baf9-49de-ab48-16bfc71e8e3a">

4. If you are using the query builder and you switch to the query editor, what is the query string that you find? This will tell us the OData string that the query builder is passing to Azure, and we can check whether the query builder is building a correct string or not.

- We have not yet tried it in query editor.
Please give us the query you would like us to run in our Azure Storage Explorer environment.

5. Are you getting any results at all if you run your query?

- Yes. Please see the image I shared. I do not see any data stored in my storage account and am told there is no data at all.

6. If you switch to the query editor and run the query, do you get any error messages?

- We have not yet tried it in query editor.
Please give us the query you would like us to run in our Azure Storage Explorer environment.

7. Does this issue also occur in 1.29.0?

- Yes. It is reproduced in 1.29.0.

**Comment by MRayermannMSFT** [comment]
@takuya-shima 

Two things....

First, can you:
1. Go to a table where you are having the issue
2. Create and run your query
3. Click on the button I have circled in orange:
![image](https://github.com/microsoft/AzureStorageExplorer/assets/21112697/a751b98a-cf24-4fef-8bbd-9fbf2f70cfb3)
4. Take a screenshot that shows the text version of the query (the textbox in the bottom portion of my screenshot)
5. Share that screenshot here

Second, would it possible for you to share a Fiddler trace with us of you reproducing the issue? You can find instructions on how to do so here: https://learn.microsoft.com/en-us/troubleshoot/azure/azure-storage/storage-explorer-troubleshooting?tabs=Windows#network-logs

We'd be looking for the trace to merely show the network requests sent when you press the run query button (the arrow button). Once you have the trace, you can put it in a .zip and then email it `sehelp` `at` `microsoft` `dot` `com`. That is a support email for our team.

Here's the request I see in Fiddler after I run a query:
![image](https://github.com/microsoft/AzureStorageExplorer/assets/21112697/bb6bdb11-2739-49ef-a812-096ad536b88b)

We'll be looking for the trace to contain at least one or more requests that look like that.

Thank you.

**Comment by takuya-shima** [comment]
@MRayermannMSFT

Thank you for your reply.
I appreciate your help and investigation.
I will first share a snapshot of the output of the execution of your requested query.
This is the output result executed by the query editor in 1.29.1.
I executed the query but nothing was output.
<img width="960" alt="20230522_AzureStorageExplorer_queryeditor" src="https://github.com/microsoft/AzureStorageExplorer/assets/104060952/d334c5aa-8765-465e-8101-f7a18dc50472">

You ask for a communication log with fidder, what is the exact procedure?

**Comment by MRayermannMSFT** [comment]
@takuya-shima you can find instructions on how to gather a fiddler trace at this link: https://learn.microsoft.com/en-us/troubleshoot/azure/azure-storage/storage-explorer-troubleshooting?tabs=Windows#network-logs

**Comment by MRayermannMSFT** [comment]
@takuya-shima if you are unable to get Fiddler working, a second option would be to share your Storage Explorer logs with us. Instructions for that can be found at this link: https://learn.microsoft.com/en-us/troubleshoot/azure/azure-storage/storage-explorer-troubleshooting?tabs=Windows#storage-explorer-logs

The Fiddler trace is preferred though.

**Comment by JoeEdwardsGitHub** [comment]
I can confirm that I am seeing the same issues. Since version 1.29.0 I get non-deterministic results from querying table storage. Mostly I get no data returned, but sometimes I get partial results. If I rerun the query multiple times I can get different results. It seems to be worst when doing queries that do not use the PartitionKey as a filter. I can sometimes (but not always) cure it by clicking Refresh followed by Execute Query followed by Load More.

I can no longer trust the results returned so I will be forced to regress to version 1.28.0 until it is fixed.

If I can provide any useful specific I will be happy to.

**Comment by craxal** [comment]
@JoeEdwardsGitHub @takuya-shima In addition to the aforementioned questions, please compare what you see in 1.28.0 and 1.29.1. Specifically:
1. Build your "Last 24 Hours" in the query builder the switch to the Query Editor. Compare the displayed query strings. Are they the same in both versions?
2. Run the query and look for the list entities request in Fiddler. Compared the URIs. Are they the same in both versions?

**Comment by JoeEdwardsGitHub** [comment]
@craxal sorry I should have clarified, I was not running a "Last 24 Hours" query, I am seeing issues with any query that I run on table storage (via ASE), but it does seem to be most prevalent with queries that do not use the standard system fields like PartitionKey and RowKey. For example I just ran a query that was "System eq 'LogAnalytics' and Database eq 'Storage' and Schema eq 'Data'" and it returns no data, even though I can see that data in the table, and if I query the table with Azure Synapse I get the data back. Also sometimes it returns data, sometimes it doesn't, sometimes it only returns some of the rows (in that particular query for example there are 220 rows, but it only returned 42). I will try to create a repeatable scenario and post it here, and try a variety of other queries as well. The only thing that I did when v1.29.0 was released was I went into settings and set the new value of "Load Batch Size" to 10000, but because I then saw this erratic behaviour I removed it to see if it was the cause, but the erratic behaviour stayed. Possibly some bad artefact was left by me setting it?

**Comment by craxal** [comment]
@JoeEdwardsGitHub @takuya-shima Another detail just occurred to me. You say that sometimes you get 0 results back when the query runs, correct? Is the "Load More" button enabled when this happens? Are you querying across many different partitions? If you click "Load More", does the result set reflect more of what you expect?

**Comment by JoeEdwardsGitHub** [comment]
@craxal sometimes clicking "Load More" makes the data show, however sometimes it doesn't. The behaviour is not deterministic. In the following screen shot I just pressed the buttons yellow (1) then blue (2) then orange (3) then yellow (4) then blue (5) then orange (6):

![image](https://github.com/microsoft/AzureStorageExplorer/assets/48823574/06c6cd4c-728b-4188-9dbd-0ebdc93bf41a)

For steps 1, 2, 3, 4 and 5 no data was shown, it only appeared after step 6

**Comment by craxal** [comment]
@JoeEdwardsGitHub OK, that seems to be consistent with my description. "Load More" is the only way to get more data. The other buttons trigger a refresh.

@JoeEdwardsGitHub @takuya-shima Can you try installing this private build to see if it resolves your issue?

Platform | Download Link
-|-
macOS ARM64 | https://cralvordtest02.blob.core.windows.net/share/1.29.1-rc.1/StorageExplorer-darwin-arm64.zip?sv=2021-10-04&st=2023-05-22T21%3A26%3A42Z&se=2023-05-29T21%3A26%3A00Z&sr=b&sp=r&sig=pGbtwmrReH9Ppd%2Bjk4mX2XdC698f8E5bZ3VK07%2FhCMw%3D
macOS x64 | https://cralvordtest02.blob.core.windows.net/share/1.29.1-rc.1/StorageExplorer-darwin-x64.zip?sv=2021-10-04&st=2023-05-22T21%3A30%3A08Z&se=2023-05-29T21%3A30%3A00Z&sr=b&sp=r&sig=o4I1my3OcggDCQPa2wS4Klw2mAhI6ECjYLO8S0fay4w%3D
Windows ia32 | https://cralvordtest02.blob.core.windows.net/share/1.29.1-rc.1/StorageExplorer-windows-ia32.exe?sv=2021-10-04&st=2023-05-22T21%3A30%3A36Z&se=2023-05-29T21%3A30%3A00Z&sr=b&sp=r&sig=NsQ9JH5I%2FCbWqyqm9C1cD4WXo63prbMY9d1tHyMQk8M%3D
Windows x64 | https://cralvordtest02.blob.core.windows.net/share/1.29.1-rc.1/StorageExplorer-windows-x64.exe?sv=2021-10-04&st=2023-05-22T21%3A31%3A06Z&se=2023-05-29T21%3A31%3A00Z&sr=b&sp=r&sig=jRrb35OKwAiXoDUQfBJF2tY31lGVTHl%2B%2FNNr%2B2sTkhA%3D

**Comment by JoeEdwardsGitHub** [comment]
@craxal thank you, that looks much better. I have only had a chance to do basic testing, but the "Execute Query" button now returns data directly and correctly, and when I change the query and click "Execute Query" again it changes the result set correctly. The "Load More" button is also working correctly.

I will be using it heavily over next few days, so will let you know if any further issues, but from my current perspective you have fixed the issue.

Thanks,

Joe

**Comment by craxal** [comment]
@JoeEdwardsGitHub Excellent, thanks for trying it out so quickly!

@takuya-shima I'll give you some time to try the private builds to see if your issue resolves. If so, we'll be sure to include the fix in the next release, which should be ready very soon.

**Comment by takuya-shima** [comment]
@craxal Thank you for your response. I have, as a matter of urgency, verified the behavior of the above new binary. I checked the behavior of the last 1 hour, the last 24 hours, and the last 7 days in timestamp on the query buider. It appears to be working correctly. I have not yet started the verification process with Fiddler.

- last 1 hour
<img width="960" alt="20230523_AzureStorageExplorer_querybuilder_1H" src="https://github.com/microsoft/AzureStorageExplorer/assets/104060952/dd5fb424-c5ab-4993-bf77-aac0942e4397">

- last 24 hours
<img width="960" alt="20230523_AzureStorageExplorer_querybuilder_24H" src="https://github.com/microsoft/AzureStorageExplorer/assets/104060952/e79bc231-6797-44ac-b2b6-94b7fcfc4fb2">

- last 7 days
<img width="960" alt="20230523_AzureStorageExplorer_querybuilder_7days" src="https://github.com/microsoft/AzureStorageExplorer/assets/104060952/16328f46-b50d-4640-bddb-c2bf5a4b219b">

**Comment by craxal** [comment]
@takuya-shima Thank you for verifying! It sounds like we've found the root cause, so we'll be working on getting that fix out right away.

**Comment by takuya-shima** [comment]
@craxal Thank you very much. We look forward to a new version with the bugs fixed.

**Comment by takuya-shima** [comment]
@MRayermannMSFT
[takuya-shima.zip](https://github.com/microsoft/AzureStorageExplorer/files/11539942/takuya-shima.zip)
I used fidder to capture the communication traffic. 
The communication I captured is the communication specified by time in timestamp on the query builder. There are four patterns.
1. the last 1 hour
2. the last 24 hours
3. the last 7 days
4. the last 31 days. 

I ran them consecutively. 
If you have other communications that need to be captured, please let us know.

**Comment by takuya-shima** [comment]
@MRayermannMSFT 

I am sorry. I have not set Azure Storage Explorer in debug mode. 
Therefore, I changed it to debug mode and captured it again. 
Please use this one.

[takuya-shima debug mode.zip](https://github.com/microsoft/AzureStorageExplorer/files/11540046/takuya-shima.debug.mode.zip)

**Comment by takuya-shima** [comment]
@MRayermannMSFT 

Here is a capture of the screen when executed by fiddler.
The version of Azure Storage Explorer is 1.29.1 which is presumed to have a bug.

- the last 1 hour

<img width="960" alt="1H" src="https://github.com/microsoft/AzureStorageExplorer/assets/104060952/559d7dc3-042d-4249-9a4f-37e272451349">

- the last 24 hours

<img width="960" alt="24H" src="https://github.com/microsoft/AzureStorageExplorer/assets/104060952/54de7305-2363-413f-934f-b60058376ca6">

- the last 7 days

<img width="960" alt="7days" src="https://github.com/microsoft/AzureStorageExplorer/assets/104060952/b712e583-4653-4e8b-96cb-499cdaa32fe1">

- the last 31 days.

<img width="960" alt="31days" src="https://github.com/microsoft/AzureStorageExplorer/assets/104060952/940b24c8-28cb-4437-a408-52347c899b1d">

**Comment by takuya-shima** [comment]
@craxal It's fine for you to close this issue, but what was the conclusion in the end?

**Comment by craxal** [comment]
@takuya-shima As I mentioned earlier, it sounds like we've found the root cause, so we'll be working on getting that fix out right away. I closed the issue, because a fix has been merged into our code base.

The root cause is, when listing items, Azure sometimes returns a result with 0 items and a continuation token. In previous versions, we tried to fill the local cache with a certain number of items (up to 1000), so if we get continuation tokens, we call Azure again with that token until we either fill the cache or Azure stops returning continuation tokens. In 1.29, we failed to carry over this behavior when reworking the table explorer, it's now been fixed and will be present in 1.29.2.

**Comment by takuya-shima** [comment]
@craxal I understand your explanation. Thank you for your quick response. We are looking forward to the release of 1.29.2. Thank you very much.

**Comment by jinglouMSFT** [comment]
We have released v1.29.2 this afternoon PDT. You should shortly receive an in-product upgrade notification. If you can't wait, you can install the bits from this Github repo. It's available already.  Thanks for your patience to help us work through this issue!  Much appreciated!

**Comment by takuya-shima** [comment]
@jinglouMSFT I am investigating the log data of the Azure storage account with the new version "1.29.2", and I can use it without any problems. Thank you for your support.
